telemetry/definitions/commonDefinitions.json

```diff
@@ -129,7 +129,7 @@
             "name": "credentialSourceId",
             "type": "string",
             "description": "Where credentials are stored or retrieved from",
-            "allowedValues": ["sharedCredentials", "sdkStore", "ec2", "envVars", "other"]
+            "allowedValues": ["sharedCredentials", "sdkStore", "ec2", "ecs", "envVars", "other"]
```

Comment:
I wonder if we should just have one generic kinda named `imds`. Because the granular type of the resolved credentials is given by `CredentialType`: https://github.com/aws/aws-toolkit-common/blob/52bfaa75321afa1bf383bd7600c8272c79e10097/telemetry/definitions/commonDefinitions.json#L135-L148